torch/csrc/jit/passes/quantization/qat_combine_conv_bn.cpp

```diff
@@ -0,0 +1,315 @@
+  scaled_weight_no_fq->addInput(conv_weight->output());
+  scaled_weight_no_fq->addInput(scale_factor_reshaped->output());
+  scaled_weight_no_fq->output()->setType(TensorType::get())
+    ->setDebugName("scaled_weight_no_fq");
+
+  // conv_bias = conv.bias
+  Node* conv_bias = g->createGetAttr(matched_conv_val, "bias");
+  conv_bias->output()->setDebugName("conv_bias");
+  g->insertNode(conv_bias);
+
+  // conv_stride = [1, 1]. TODO: get from conv instead of hardcode
+  Node* conv_stride = g->create(prim::ListConstruct);
+  conv_stride->addInput(one->output());
+  conv_stride->addInput(one->output());
+  conv_stride->output()->setType(ListType::create(IntType::get()))
+    ->setDebugName("conv_stride");
+  g->insertNode(conv_stride);
+
+  // conv_padding = [0, 0]. TODO: get from conv instead of hardcode
+  Node* conv_padding = g->create(prim::ListConstruct);
+  conv_padding->addInput(zero->output());
+  conv_padding->addInput(zero->output());
+  conv_padding->output()->setType(ListType::create(IntType::get()))
+    ->setDebugName("conv_padding");
+  g->insertNode(conv_padding);
+
+  // conv_dilation = [1, 1]. TODO: get from conv instead of hardcode
+  Node* conv_dilation = g->create(prim::ListConstruct);
+  conv_dilation->addInput(one->output());
+  conv_dilation->addInput(one->output());
+  conv_dilation->output()->setType(ListType::create(IntType::get()))
+    ->setDebugName("conv_dilation");
+  g->insertNode(conv_dilation);
+
+  // conv_res = aten::qat_conv2d_and_unscale(input, scaled_weight_no_fq,
+  //   scale_factor, conv_bias, conv_stride, conv_padding, conv_dilation,
+  //   one)
+  Node* conv_res = g->insertNode(g->create(aten::qat_conv2d_and_unscale));
+  Value* input = g->inputs()[1];
+  conv_res->addInput(input);
+  conv_res->addInput(scaled_weight_no_fq->output());
+  conv_res->addInput(scale_factor->output());
+  conv_res->addInput(conv_bias->output());
+  conv_res->addInput(conv_stride->output());
+  conv_res->addInput(conv_padding->output());
+  conv_res->addInput(conv_dilation->output());
+  conv_res->addInput(one->output());
+  conv_res->output()->setDebugName("conv_res");
+
+  // replace bn forward's tensor input with conv_res
+  matched_bn_orig_forward_call->replaceInput(1, conv_res->output());
+
+  // TODO: delete the original conv forward
+
+  // TODO: verify numerics match
+
+  GRAPH_DUMP("after", g);
+
+}
+
+void QATCombineConvBatchNorm2dHelper::transform(Module& module) {
+
+  // Dot in the ".Conv2d" and ".BatchNorm2d" is an attempt to
+  // prevent matching module's whose name might end with Conv2d
+  // But are user defined modules.
+  const PatternInfo pattern = PatternInfo::parse_from_str(R"IR(
+graph(%self, %x):
+    %conv_submodule = match::module[name=".Conv2d"](%self)
+    %conv_out = prim::CallMethod[name="forward"](%conv_submodule, %x)
+    %bn_submodule = match::module[name=".BatchNorm2d"](%self)
+    %bn_out = prim::CallMethod[name="forward"](%bn_submodule, %conv_out)
+    return (%bn_out))IR");
```

Comment:
btw we have different way of matching now: https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/fold_conv_bn.cpp.html#358, this might help with expressing the transformation with subgraph_rewriter as well